connectors/src/connectors/slack/index.ts

```diff
@@ -427,17 +455,49 @@ export async function setSlackConnectorPermissions(
 
   let shouldGarbageCollect = false;
   for (const [id, permission] of Object.entries(permissions)) {
-    const channel = channels[id];
+    let channel = channels[id];
+    const slackAccessToken = await getAccessToken(connector.connectionId);
+    const slackClient = await getSlackClient(slackAccessToken);
     if (!channel) {
-      logger.error(
-        { connectorId, channelId: id },
-        "Could not find channel in DB"
-      );
-      continue;
+      const remoteChannel = await slackClient.conversations.info({
+        channel: id,
+      });
+      if (!remoteChannel.ok || !remoteChannel.channel?.name) {
+        logger.error(
+          {
+            connectorId,
+            channelId: id,
+            error: remoteChannel.error,
+          },
+          "Could not get the Slack channel information"
+        );
+        return new Err(
+          new Error("Could not get the Slack channel information.")
+        );
+      }
+      const joinRes = await joinChannel(connectorId, id);
+      if (joinRes === "cant_join") {
```

Comment:
Here the design is a bit brittle. You silently accept if it's anyting but cant_join but if I go and change the signature of joinChannel to add a new case then this code become erroneous. Let's change the joinChannel signature to a more classic Result<Channel, TypedError> and handle the TypedError.reason or code or type cases explicitely

Reply:
Agree we should handle all the cases, but it is a "standard" in our code base to never use Result<> in activities.ts files because they won't properly serialize / deserialize over Temporal.

We need a lib that returns a result and an activity that throws then.

Reply:
But we're within activity here?

Reply:
Initially this was only called from the activity, and I then called it from the permission endpoint, which ended up being an odd design. I will fix it right now.